pkg/config/validation/validation.go

```diff
@@ -763,11 +763,42 @@ var ValidateSidecar = registerValidateFunc("ValidateSidecar",
 					errs = appendErrors(errs, validateNamespaceSlashWildcardHostname(hostname, false))
 				}
 			}
+
 		}
 
+		errs = appendErrors(errs, validateSidecarOutboundTrafficPolicy(rule.OutboundTrafficPolicy))
+
 		return
 	})
 
+func validateSidecarOutboundTrafficPolicy(tp *networking.OutboundTrafficPolicy) (errs error) {
+	if tp == nil {
+		return
+	}
+	mode := tp.GetMode()
+	if tp.EgressProxy != nil {
+		if mode != networking.OutboundTrafficPolicy_ALLOW_ANY {
+			errs = appendErrors(errs, fmt.Errorf("sidecar: egress_proxy must be set only with ALLOW_ANY outbound_traffic_policy mode"))
+			return
+		}
+
+		errs = appendErrors(errs, validateSidecarOutboundEgressProxy(tp.EgressProxy))
+	}
+	return
+}
+
+func validateSidecarOutboundEgressProxy(ep *networking.Destination) (errs error) {
+	if ep == nil {
```

Comment:
this cannot happen since you are already checking for non nil. I would suggest you fold this function into the previous function